src/Arcus.EventGrid.Tests.Integration/Publishing/CloudEventPublishing.cs

```diff
@@ -0,0 +1,196 @@
+        }
+
+        private void AssertReceivedNewCarRegisteredEventWithTimeout(CloudEvent @event, string licensePlate)
+        {
+            TracePublishedEvent(@event.Id, @event);
+            string receivedEvent = _endpoint.ServiceBusEventConsumerHost.GetReceivedEvent(@event.Id, timeout: TimeSpan.FromSeconds(30));
+            AssertReceivedNewCarRegisteredEvent(@event.Id, @event.Type, @event.Subject, licensePlate, receivedEvent);
+        }
+
+        [Fact]
+        public async Task PublishSingleRawEvent_WithBuilder_ValidParameters_Succeeds()
+        {
+            // Arrange
+            const string licensePlate = "1-TOM-337";
+            const string expectedSubject = "/";
+            var eventId = Guid.NewGuid().ToString();
+            var data = new CarEventData(licensePlate);
+            var rawEventBody = JsonConvert.SerializeObject(data);
+            var cloudEvent = new CloudEvent(CloudEventsSpecVersion.V1_0, "NewCarRegistered", new Uri("http://test-host"), subject: expectedSubject, id: eventId)
+            {
+                Data = data,
+                DataContentType = new ContentType("application/json")
+            };
+
+            IEventGridPublisher publisher = _endpoint.BuildPublisher();
+
+            // Act
+            await publisher.PublishRawCloudEventAsync(cloudEvent.SpecVersion, cloudEvent.Id, cloudEvent.Type, cloudEvent.Source, rawEventBody);
+            TracePublishedEvent(eventId, cloudEvent);
+
+            // Assert
+            var receivedEvent = _endpoint.ServiceBusEventConsumerHost.GetReceivedEvent(eventId);
+            AssertReceivedNewCarRegisteredEvent(eventId, cloudEvent.Type, expectedSubject, licensePlate, receivedEvent);
+        }
+
+        [Fact]
+        public async Task PublishSingleRawEventWithDetailedInfo_WithBuilder_ValidParameters_Succeeds()
+        {
+            // Arrange
+            const string licensePlate = "1-TOM-337";
+            const string expectedSubject = "/";
+            var eventId = Guid.NewGuid().ToString();
+            var data = new CarEventData(licensePlate);
+            var rawEventBody = JsonConvert.SerializeObject(data);
+            var cloudEvent = new CloudEvent(CloudEventsSpecVersion.V1_0, "NewCarRegistered", new Uri("http://test-host"), subject: expectedSubject, id: eventId, DateTime.UtcNow)
+            {
+                Data = data,
+                DataContentType = new ContentType("application/json")
+            };
+
+            IEventGridPublisher publisher = _endpoint.BuildPublisher();
+
+            // Act
+            await publisher.PublishRawCloudEventAsync(
+                cloudEvent.SpecVersion,
+                cloudEvent.Id,
+                cloudEvent.Type,
+                cloudEvent.Source,
+                cloudEvent.Subject,
+                rawEventBody,
+                cloudEvent.Time ?? default(DateTimeOffset));
```

Comment:
We control this so why do we need to check this?

Reply:
Bc `CloudEvent.Time` is a `Nullable<DateTime>`.